test/aeso_compiler_tests.erl

```diff
@@ -218,6 +218,9 @@ compilable_contracts() ->
      "using_namespace",
      "assign_patterns",
      "patterns_guards",
+     "contract_polymorphism",
+     "contract_polymorphism_multi_interface",
+     "contract_interface_polymorphism",
```

Comment:
Add tests for custom datatypes, records and type aliases. Please consider various variances. Also there should be tests for tuples, maps, lists and oracles.

Reply:
In the files `polymorphism_variance_switching.aes` and `polymorphism_variance_switching_custom_types.aes` there are tests for:

* custom datatypes
* tuples
* maps
* lists

I'll add to that tests for:
* options
* oracles
* oracle queries
* records
* type aliases

Reply:
I have tests for all the above now after 114010b8e1ff9a978912f736bd143364666bd89c, 8e995429aa6bd184b3d690f97dacf3ea6fb1b555, eab8eae9c1b97174fa93cb31201805aa45a4a578, and a6f1d8107713baa0379ecbacd06c1e51b4bd112c.